client/src/models/Test/Test.ts

```diff
@@ -54,9 +54,22 @@ class Test {
     this.store.set(this.getComponentFormat());
   }
 
+  calculateScore(): number {
+    let score = 0;
+    this.questions.forEach(question => {
+      let MCquestion = question as MultipleChoiceQuestion;
+      if (MCquestion.selectedAnswer == MCquestion.correctAnswer) {
```

Comment:
It would be better to have this line be a method in the MultipleChoiceQuestion called isCorrect() so that not too much responsibility is put on the calculateScore method.